src/Connection/RelayConnection.php

```diff
@@ -198,9 +199,11 @@ public function getClient()
     /**
      * {@inheritdoc}
      */
-    protected function getIdentifier()
+    public function getIdentifier()
     {
-        return $this->client->endpointId();
+        // Using `spl_object_id()` workaround until `Relay::endpointId()`
+        // works without requiring a connection
+        return (string) spl_object_id($this->client);
```

Comment:
This will be more accurate.

```suggestion
        try {
            return $this->client->endpointId();
        } catch (RelayException $ex) {
            return parent::getIdentifier();
        }
```